boolean step type failure (expected str, bytes or os.PathLike object, not bool)

env:
```bash
$ which conjure-up
/snap/bin/conjure-up

$ conjure-up --version
conjure-up 2.5-alpha1

$ snap info conjure-up
name:      conjure-up
...
tracking:    edge
installed:   2.5-alpha1-20171201.1512 (866) 92MB classic
refreshed:   2017-12-01 09:29:57 -0600 CST
channels:                                   
  stable:    2.4.2-20171130.1811      (859) 82MB classic
  candidate: 2.4.2-20171130.1811      (859) 82MB classic
  beta:      2.4.2-20171130.1811      (859) 82MB classic
  edge:      2.5-alpha1-20171201.1512 (866) 92MB classic
```

I'm trying to use a boolean step type to present an option to my users. The metadata looks like this, ./steps/05_logging/metadata.yaml:
```yaml
title: Logging Mechanism
description: Optional logging components
required: True
viewable: True
additional-input:
  - label: Elasticsearch/Filebeat/Graylog
    key: LOGGINGPLUGIN
    type: boolean
    default: "No"
```
And the `after-input` looks like this:

```bash
#!/bin/bash

set -eux

. "$CONJURE_UP_SPELLSDIR/sdk/common.sh"

if [[ "$LOGGINGPLUGIN" == "Yes" ]]; then
    setStepKey "bundle-add" "efg.yaml"
fi

exit 0
```
In the GUI, I leave the default "No" checked when prompted by the step, and click next.  I see this:

![screen shot 2017-12-01 at 10 39 37 am](https://user-images.githubusercontent.com/4576822/33492876-f5c3a2f6-d683-11e7-968c-f2ae5d598551.png)

Good news is that #1254 is clearly fixed, as it shows the correct log path now.  Super!

Bad news is that it fails like this:

```
2017-12-01 10:27:56,803 [DEBUG] conjure-up/canonical-kubernetes - telemetry.py:17 - Showing screen: Configure Spell
2017-12-01 10:27:58,238 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-12-01 10:27:58,239 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Text flow widget ''>, ('weight', 1))
2017-12-01 10:27:58,239 [INFO] conjure-up/canonical-kubernetes - step.py:205 - Running step: 01_select-network after-input.
2017-12-01 10:27:58,448 [DEBUG] conjure-up/canonical-kubernetes - step.py:242 - Storing environment
2017-12-01 10:27:58,457 [DEBUG] conjure-up/canonical-kubernetes - step.py:248 - Executing script: /tmp/test/canonical-kubernetes/steps/01_select-network/after-input
2017-12-01 10:27:59,319 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-12-01 10:27:59,330 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Text flow widget ''>, ('weight', 1))
2017-12-01 10:28:04,662 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Padding selectable flow widget <AttrMap selectable flow widget <PlainButton selectable flow widget 'Next'> attr_map={None: 'button_primary'} focus_map={None: 'button_primary focus'}> align='right' width=('relative', 20)>, ('weight', 1))
2017-12-01 10:28:04,663 [DEBUG] conjure-up/canonical-kubernetes - step.py:101 - Contents: (<Text flow widget ''>, ('weight', 1))
2017-12-01 10:28:04,664 [INFO] conjure-up/canonical-kubernetes - step.py:205 - Running step: 05_logging after-input.
2017-12-01 10:28:04,873 [DEBUG] conjure-up/canonical-kubernetes - step.py:242 - Storing environment
2017-12-01 10:28:04,878 [DEBUG] conjure-up/canonical-kubernetes - step.py:248 - Executing script: /tmp/test/canonical-kubernetes/steps/05_logging/after-input
2017-12-01 10:28:04,882 [DEBUG] conjure-up/canonical-kubernetes - events.py:53 - Setting Error at conjureup/events.py:148
2017-12-01 10:28:04,882 [ERROR] conjure-up/canonical-kubernetes - events.py:160 - Unhandled exception in <Task finished coro=<ShowStepsController.show_steps() done, defined at /snap/conjure-up/866/lib/python3.6/site-packages/conjureup/controllers/showsteps/gui.py:22> exception=TypeError('expected str, bytes or os.PathLike object, not bool',)>
Traceback (most recent call last):
  File "/snap/conjure-up/866/lib/python3.6/site-packages/conjureup/controllers/showsteps/gui.py", line 30, in show_steps
    await self.show_step(step)
  File "/snap/conjure-up/866/lib/python3.6/site-packages/conjureup/controllers/showsteps/gui.py", line 42, in show_step
    await step.after_input(app.ui.set_footer)
  File "/snap/conjure-up/866/lib/python3.6/site-packages/conjureup/models/step.py", line 105, in after_input
    return await self.run(PHASES.AFTER_INPUT, msg_cb)
  File "/snap/conjure-up/866/lib/python3.6/site-packages/conjureup/models/step.py", line 255, in run
    cb_stdout=msg_cb)
  File "/snap/conjure-up/866/lib/python3.6/site-packages/conjureup/utils.py", line 162, in arun
    **kwargs)
  File "/snap/conjure-up/866/usr/lib/python3.6/asyncio/subprocess.py", line 225, in create_subprocess_exec
    stderr=stderr, **kwds)
  File "/snap/conjure-up/866/usr/lib/python3.6/asyncio/base_events.py", line 1191, in subprocess_exec
    bufsize, **kwargs)
  File "/snap/conjure-up/866/usr/lib/python3.6/asyncio/unix_events.py", line 191, in _make_subprocess_transport
    **kwargs)
  File "/snap/conjure-up/866/usr/lib/python3.6/asyncio/base_subprocess.py", line 39, in __init__
    stderr=stderr, bufsize=bufsize, **kwargs)
  File "/snap/conjure-up/866/usr/lib/python3.6/asyncio/unix_events.py", line 697, in _start
    universal_newlines=False, bufsize=bufsize, **kwargs)
  File "/snap/conjure-up/866/usr/lib/python3.6/subprocess.py", line 707, in __init__
    restore_signals, start_new_session)
  File "/snap/conjure-up/866/usr/lib/python3.6/subprocess.py", line 1247, in _execute_child
    env_list.append(k + b'=' + os.fsencode(v))
  File "/snap/conjure-up/866/usr/lib/python3.6/os.py", line 798, in fsencode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not bool
2017-12-01 10:28:04,900 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:24 - Showing dialog for exception: expected str, bytes or os.PathLike object, not bool
```

Full log at https://pastebin.canonical.com/204645/

--- a/conjureup/models/step.py
+++ b/conjureup/models/step.py
@@ -231,7 +231,7 @@ class StepModel:
 
         for step_name, step_data in app.steps_data.items():
             for key, value in step_data.items():
-                app.env[key.upper()] = step_data[key]
+                app.env[key.upper()] = str(step_data[key])
 
         for key, value in app.env.items():
             if value is None:

